From: Jeff Moyer <jmoyer@redhat.com>
Date: Mon, 8 Sep 2008 13:49:27 -0400
Subject: [scsi] sd: revalidate_disk wrapper
Message-id: x493akav87s.fsf@segfault.boston.devel.redhat.com
O-Subject: [rhel5 patch 4/6 - v2] SCSI sd driver calls revalidate_disk wrapper.
Bugzilla: 444964
RH-Acked-by: Josef Bacik <jbacik@redhat.com>
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>
RH-Acked-by: Josef Bacik <jbacik@redhat.com>
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

SCSI sd driver calls revalidate_disk wrapper.

Modify the SCSI disk driver to call the revalidate_disk()
wrapper. This allows us to do some housekeeping such as accounting for
a disk being resized online. The wrapper will call
sd_revalidate_disk() at the appropriate time.

Signed-off-by: Andrew Patterson <andrew.patterson@hp.com>

diff --git a/drivers/scsi/sd.c b/drivers/scsi/sd.c
index b4ef996..525864b 100644
--- a/drivers/scsi/sd.c
+++ b/drivers/scsi/sd.c
@@ -214,7 +214,7 @@ static ssize_t sd_store_cache_type(struct class_device *cdev, const char *buf,
 			scsi_print_sense_hdr(sdkp->disk->disk_name, &sshdr);
 		return -EINVAL;
 	}
-	sd_revalidate_disk(sdkp->disk);
+	revalidate_disk(sdkp->disk);
 	return count;
 }
 
@@ -888,7 +888,7 @@ static void sd_rescan(struct device *dev)
 	struct scsi_disk *sdkp = scsi_disk_get_from_dev(dev);
 
 	if (sdkp) {
-		sd_revalidate_disk(sdkp->disk);
+		revalidate_disk(sdkp->disk);
 		scsi_disk_put(sdkp);
 	}
 }
